tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Serializers/KnownSerializers/KnownSerializerFinderTests.cs

```diff
@@ -0,0 +1,228 @@
+/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq.Expressions;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Linq.Linq3Implementation.Serializers.KnownSerializers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Serializers.KnownSerializers
+{
+    public class KnownSerializerFinderTests
+    {
+        private readonly IBsonDocumentSerializer _collectionSerializer = (IBsonDocumentSerializer)BsonSerializer.LookupSerializer<C>();
+
+        #region class definitions for testing
+        private enum E { A, B }
+
+        private class C
+        {
+            public int P { get; set; }
+            [BsonRepresentation(BsonType.Int32)]
+            public E Ei { get; set; }
+            [BsonRepresentation(BsonType.String)]
+            public E Es { get; set; }
+            public A A { get; set; }
+        }
+
+        private class A
+        {
+            public B B { get; set; }
+        }
+
+        private class B { }
+
+        private class View
+        {
+            public A A { get; set; }
+        }
+        #endregion
+
+        [Fact]
+        public void Identity_expression_should_return_collection_serializer()
+        {
+            Expression<Func<C, C>> expression = x => x;
+
+            var result = KnownSerializerFinder.FindKnownSerializers(expression, _collectionSerializer);
+
+            var serializer = result.GetSerializer(expression.Body);
+            serializer.Should().Be(_collectionSerializer);
+            serializer.Should().BeOfType<BsonClassMapSerializer<C>>();
+        }
+
+        [Fact]
+        public void Int_property_expression_should_return_int_serializer()
+        {
+            Expression<Func<C, int>> expression = x => x.P;
+
+            var result = KnownSerializerFinder.FindKnownSerializers(expression, _collectionSerializer);
+
+            var serializer = result.GetSerializer(expression.Body);
+            _collectionSerializer.TryGetMemberSerializationInfo(nameof(C.P), out var expectedPropertySerializer);
```

Comment:
I understand we are not testing the `TryGetMemberSerializationInfo` method and hence your reluctance to use `Should().BeTrue()` as follows:

    collectionSerializer.TryGetMemberSerializationInfo(nameof(C.P), out var expectedPropertySerializationInfo).Should().BeTrue();

However... we are *assuming* this returns `true`. Consider what happens if due to some unforeseen circumstance it should return `false`:

1. Hopefully `expectedPropertySerializationInfo` will be set to null and we will get a `NullReferenceException` on the following line
2. Worst case: `expectedPropertySerializationInfo` is set to some random serializer and the assert on the following line fails and we tear our hair out trying to figure out why

Therefore I think we really should add `Should().BeTrue()`.

Reply:
Also... rename out var to `expectedPropertySerializationInfo`

Reply:
Similarly for all other tests in this file.